hs/WebParsing/PrerequisiteParsing.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{-|
+module          : WebParsing.PrerequisiteParsing
+Description     : converts a T.Text representation of prerequisites into prereq format
+Stability       : experimental  
+
+Currently parses prerequisites for Arts & Science only. Attempts to preserve only the most
+basic internal logic of the prerequisite representation found on course calendar: each element
+is either:  1) a 1-element list containig a course name 
+            2) an n-element list containing a list of course names.
+-}
+module WebParsing.PrerequisiteParsing (parsePrerequisites) where 
+
+import Text.Regex.Posix
+import qualified Data.Text as T
+import Data.List
+import Data.List.Utils
+import Data.Maybe
+
+{- Signatures:
+
+-}
+
+-- | attempts to match characters used to delimit prerequisite expressions
+-- returns (before, delim, after). or (input, "","") if no match occurs
+matchDelim :: String -> (String, String, String)
+matchDelim prereqs = 
+  let pat = "[;,]" :: String
+  in prereqs =~ pat
+      
+-- | returns true if the string begins inside a parenthesized expression.
+-- e.g  isntDelim "CSC458)" == True
+--      isntDelim "(STA247, STA248)" == False
+isntDelim :: String -> Bool
+isntDelim rest = 
+  let pat = "^.[^(]*\\)" :: String
```

Comment:
Er, I'm not sure why you made this change - it seems quite different from the original.